src/python-common/ceph/deployment/translate.py

```diff
@@ -25,27 +25,38 @@ def __init__(self,
         self.osd_id_claims = osd_id_claims
 
     def run(self):
-        # type: () -> Optional[str]
+        # type: () -> List[str]
         """ Generate ceph-volume commands based on the DriveGroup filters """
         data_devices = [x.path for x in self.selection.data_devices()]
         db_devices = [x.path for x in self.selection.db_devices()]
         wal_devices = [x.path for x in self.selection.wal_devices()]
         journal_devices = [x.path for x in self.selection.journal_devices()]
 
         if not data_devices:
-            return None
+            return []
 
-        cmd = ""
+        cmds: List[str] = []
         if self.spec.method == 'raw':
             assert self.spec.objectstore == 'bluestore'
-            cmd = "raw prepare --bluestore"
-            cmd += " --data {}".format(" ".join(data_devices))
-            if db_devices:
-                cmd += " --block.db {}".format(" ".join(db_devices))
-            if wal_devices:
-                cmd += " --block.wal {}".format(" ".join(wal_devices))
+            # ceph-volume raw prepare only support 1:1 ratio of data to db/wal devices
+            if data_devices and db_devices:
+                assert len(data_devices) == len(db_devices)
```

Comment:
Minor but: I prefer seeing `assert` only in test code. It's rare but it is possible to turn off assertions in python. I also prefer that prod. code raise ValueError over AssertionErrors. but ymmv